athena/active.py

```diff
@@ -318,3 +324,23 @@ def _rotate_x(self, reduced_inputs, inactive_inputs):
         inputs = np.dot(YZ, self.evects.T).reshape((N * NY, m))
         indices = np.kron(np.arange(NY), np.ones(N)).reshape((N * NY, 1))
         return inputs, indices
+
+    def _frequent_directions(self, gradients):
+        """
+        Function that perform the frequent direction algorithm.
+        :param numpy.ndarray inputs: input parameters oriented as columns
+        :param generator
+        :return tensor V and sigma: matrix of projection V and singular
+            values sigma
+        """
+        s = []
+        for i in range(self.dim):
+            #s[:, i] = next(gradients)
+            s.extend([next(gradients)])
+        s = np.array(s).T
+        for grad in gradients:
+            v, sigma, uh = np.linalg.svd(s, full_matrices=False)
```

Comment:
you are not using uh so you can save only the first two variables with for example [:2] at the end